Wizualizacja gry

NintyS opened the pull request:
Jest poruszanie czołgiem, jest przeszkoda, nie ma kolizji.

NintyS committed:
Wizualizacja gry

Jest poruszanie czołgiem, jest przeszkoda, nie ma kolizji.

wrazik commented on Wizualizacja gry/Obstacle/ObstacleClass.hpp:
```diff
@@ -0,0 +1,29 @@
+#define ObstacleClass_hpp
+
+#include <SFML/Graphics.hpp>
+#include <stdio.h>
```
Don't use C headers. Use `<cstdio>`

wrazik commented on Wizualizacja gry/Tank/TankClass.cpp:
```diff
@@ -0,0 +1,44 @@
+    
+}
+
+void Tank::move(std::string direction) {
```
Create `enum class` with allowed directions:
```
enum class Direction {
    Up,
   Down,
   Left,
   Right
};
```

wrazik commented on Wizualizacja gry/Tank/TankClass.cpp:
```diff
@@ -0,0 +1,44 @@
+}
+
+sf::Sprite Tank::getSprite() {
+    return sprite;
```
You are returning the copy of the object - is it intentional? Maybe it would be better to return the (const) reference of the object?

wrazik commented on Wizualizacja gry/TankBotFight-Info.plist:
```diff
@@ -0,0 +1,26 @@
+	<key>NSHighResolutionCapable</key>
+	<true/>
+</dict>
+</plist>
```
Why we need this file?

wrazik commented on Wizualizacja gry/TankBotFight.entitlements:
```diff
@@ -0,0 +1,8 @@
+	<key>com.apple.security.cs.disable-library-validation</key>
+	<true/>
+</dict>
+</plist>
```
Why we need this file?

wrazik commented on Wizualizacja gry/main.cpp:
```diff
@@ -0,0 +1,78 @@
+//Helping Libraries For SFML ( MacOS )
+#include "ResourcePath.hpp"
+
+int main(int, char const**)
```
you don't need arguments here.

wrazik commented on Wizualizacja gry/main.cpp:
```diff
@@ -0,0 +1,78 @@
+            }
+        }
+        
+        /*
```
Let's not keep unused comments in repository

NintyS replied:
This code will be used, but in another form. I had to test if it move accordingly. In the future this will be getting information in which direction to move and move an object.

wrazik replied:
I believe this snippet is so simple that we don't have to store it in the repo. I would say keeping commented code in the repository is usually a bad practice. It's not adding any value to this file and just decrease the readability.

wrazik commented on Wizualizacja gry/Obstacle/ObstacleClass.hpp:
```diff
@@ -0,0 +1,29 @@
+//  Copyright © 2021 Emil Panecki. All rights reserved.
+//
+
+#ifndef ObstacleClass_hpp
```
Maybe it would be better to use 
```
 #pragma once
 ```
 ? It's the same but 1 line instead of 3

NintyS replied:
For me no difference.

wrazik commented on Wizualizacja gry/ResourcePath.mm:
```diff
@@ -0,0 +1,52 @@
+    [pool drain];
+
+    return rpath;
+}
```
wtf is this file?

wrazik commented on Wizualizacja gry/ResourcePath.hpp:
```diff
@@ -0,0 +1,43 @@
+/// with the main bundle or an empty string is there is no bundle.
+///
+////////////////////////////////////////////////////////////
+std::string resourcePath(void);
```
Why we need this file? Can we just hardcode some path instead of terribly complicated methods?

wrazik reviewed (changes requested):
We don't need folder named "Wizualizacja gry". I believe we want to have code structure like this:
```
src/
Obstacle.hpp
Obstacle.cpp
...
test/
...here will be tests
```
Why do we need .DS_Store files?

I believe we should add CMakeLists.txt from the beginning.

The same with .clang-format

Code should be reformatted according to style agreed.

NintyS committed:
Corrected

I added a src, test and res folder and repair my mistakes from previous commit.

NintyS commented:
Now when I corrected my mistakes I must send new request or what?
